orte/mca/ess/hnp/ess_hnp_module.c

```diff
@@ -462,12 +462,14 @@ static int rte_init(void)
     if (orte_retain_aliases) {
         aliases = NULL;
         opal_ifgetaliases(&aliases);
-        /* add our own local name to it */
-        opal_argv_append_nosize(&aliases, orte_process_info.nodename);
-        aptr = opal_argv_join(aliases, ',');
+        if (0 < opal_argv_count(aliases)) {
+            /* add our own local name to it */
+            opal_argv_append_nosize(&aliases, orte_process_info.nodename);
+            aptr = opal_argv_join(aliases, ',');
+            orte_set_attribute(&node->attributes, ORTE_NODE_ALIAS, ORTE_ATTR_LOCAL, aptr, OPAL_STRING);
+            free(aptr);
+        }
```

Comment:
I'm a little confused by this logic:

* If there are aliases, then set `ORTE_NODE_ALIAS` to a comma-delimited list of aliases *plus* our local nodename.
* If there are *no* aliases, then do not set `ORTE_NODE_ALIAS` at all -- not even to our local nodename.

Is the behavior w.r.t. the local nodename intended?  I.e., is the local nodename only relevant if there are aliases?

Reply:
yes - that is the intended behavior. Note that this also only happens if ```orte_retain_aliases``` is set. Otherwise, we just ignore the aliases.

Reply:
Ok.  👍